adi/lidar.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2019 Analog Devices, Inc.
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#     - Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     - Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in
+#       the documentation and/or other materials provided with the
+#       distribution.
+#     - Neither the name of Analog Devices, Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#     - The use of this software may or may not infringe the patent rights
+#       of one or more patent holders.  This license does not release you
+#       from the requirement that you obtain separate licenses from these
+#       patent holders to use this software.
+#     - Use of the software either in source or binary form, must be run
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from adi.ad5627 import ad5627
+from adi.ad9094 import ad9094
+from adi.rx_tx import phy
+import numpy as np
+
+class Lidar (ad5627, ad9094, phy):
+    """ LiDAR """
+    _device_name = "LiDAR"
+
+    def __init__(self, uri, pulse_capture_address="7c700000"):
+        ad5627.__init__(self, uri)
+        ad9094.__init__(self, uri)
+        # The name of the pulse capture channel. The address might be different,
+        # depending on how the board was built.
+        self.pulse_capture = pulse_capture_address + "." + "axi-pulse-capture"
+        self._ctrl = self._ctx.find_device(self.pulse_capture)
+        self.set_all_iio_attrs_to_default_values()
+
+    def read_channels(self):
```

Comment:
To keep uniformity across the classes it might be better to simply rename this method rx, so it won't be called by mistake. What is the purpose of the zeroed out channel? Would it ever be useful for the user to have access too?

Reply:
I think that this change will transform this function into a recursive one, since I'm calling `self.rx()` in its body.

Reply:
That channel is only relevant to HW and not to the user. In short,

> Would it ever be useful for the user to have access too?

No

Reply:
Use the [super](https://docs.python.org/3/library/functions.html#super) method to call the parent class's rx.

Reply:
Neat! It did the trick :)